cmd/bpf2go/platform.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import "runtime"
+
+func getTargetArch() string {
+	if runtime.GOOS != "linux" {
```

Comment:
runtime.GOOS is the platform we're running on, not the target platform. This would effectively prevent cross compiling BPF for Linux on e.g. a MacOS machine. I think this should be removed.